tracking/inc/track.h

```diff
@@ -15,28 +17,153 @@ enum TrackStatus { alive, dead };
 
 struct track {
   RanluxppDouble rng_state;
-  int index{0};
-  int pdg{0};
-  double energy{10};
+  unsigned int   index{0};
+  double         energy{10};
+   
   vecgeom::Vector3D<double> pos;
   vecgeom::Vector3D<double> dir;
-  vecgeom::NavStateIndex current_state;
-  vecgeom::NavStateIndex next_state;
-  int mother_index{0};
-  TrackStatus status{alive};
-  int current_process{0};
+  vecgeom::NavStateIndex    current_state;
+  vecgeom::NavStateIndex    next_state;
+
+  unsigned int mother_index{0};
+  unsigned int eventId{0};
+  TrackStatus  status{alive};
+
   float interaction_length{FLT_MAX};
-  float energy_loss{0};         // primitive version of scoring
-  int number_of_secondaries{0}; // primitive version of scoring
+  float total_length{0.0};        // length since track start
+   
+  float energy_loss{0};           // primitive version of scoring
+  int   number_of_secondaries{0}; // primitive version of scoring
+
+  uint16_t    num_step{0}; // number of Steps -- for killing after MaX
+   
+  char        current_process{0}; // -1 for Navigation / -2 for Field Propagation
+  char        pdg{0};             // Large enough for e/g. Then uint16_t for p/n
+
+  __device__ __host__ double uniform() { return rng_state.Rndm(); }
+
+  __device__ __host__ int    charge() const
+  { int chrg= (pdg== -11)-(pdg==11); return chrg; }
```

Comment:
😄

I think this deserves a comment. Or you can write it with `if`s / ternaries for now and let the compiler take care of optimizing it.

Reply:
I thought that it was adequate for the time being.  With only electrons and positrons tracked, it does not need to be complicated.

OK?

Reply:
Sure, it's ok for now, I just found the clever conversions from boolean values from comparisons to integer values of a charge remarkably clever :+1: 

The compiler probably emits the same code for `(pdg == -11) ? 1 : ((pdg == 11) ? -1 : 0)`, up to you which route to go (maybe constants for pdg numbers of electrons and positrons help, but maybe that's just me coming from a CS background)

Reply:
I agree that the code is made more readable with "constexpr pdgElec= 11".  I am adopting this.